resources/assets/lib/beatmap-discussions/user-card.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import UserAvatar from 'components/user-avatar';
+import UserGroupBadge from 'components/user-group-badge';
+import UserGroupJson from 'interfaces/user-group-json';
+import UserJson from 'interfaces/user-json';
+import { route } from 'laroute';
+import * as React from 'react';
+
+const bn = 'beatmap-discussion-user-card';
+
+interface Props {
+  group: UserGroupJson;
+  hideStripe: boolean;
+  user: UserJson;
+}
+
+export class UserCard extends React.PureComponent<Props> {
+  static readonly defaultProps = {
+    hideStripe: false,
+  };
+
+  render() {
+    return (
+      <div className={bn} style={osu.groupColour(this.props.group)}>
+        <div className={`${bn}__avatar`}>
+          {this.props.user.is_deleted ? (
+            <span className={`${bn}__user-link`}>
+              <UserAvatar modifiers='full-rounded' user={this.props.user} />
+            </span>
+          ) : (
+            <a
+              className={`${bn}__user-link`}
+              href={route('users.show', { user: this.props.user.id })}
+            >
+              <UserAvatar modifiers='full-rounded' user={this.props.user} />
+            </a>
+          )}
+        </div>
+
+        <div className={`${bn}__user`}>
+          <div className={`${bn}__user-row`}>
+            {this.props.user.is_deleted ? (
+              <span className={`${bn}__user-text u-ellipsis-overflow`}>
+                {this.props.user.username}
+              </span>
+            ) : (
+              <a
+                className={`${bn}__user-link`}
+                href={route('users.show', { user: this.props.user.id })}
+              >
+                {this.props.user.username}
```

Comment:
missing `u-ellipsis-overflow` which apparently stopped working sometime when the parent stopped being a flex container...

also missing `__user-text` so the styling is rather wrong (although now I looked closer the text and hover underline have different colour 👀 maybe fix that as well while at it).